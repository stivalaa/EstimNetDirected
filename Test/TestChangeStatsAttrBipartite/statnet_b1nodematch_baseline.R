#!/usr/bin/Rscript
##
## File:    statnet_b1nodematch_baseline.R
## Author:  Alex Stivala
## Created: December 2023
##
## Get b1nodematch and b2nodematch statistics for test data using statnet
## as gold standard to verify implementations changeBipartiteNodematchBetaA,
## changeBipartiteNodematchBetaB, changeBipartiteNodematchAlphaA, and
## changeBipartiteNodematchAlphaB in EstimNetDirected.
##
library(igraph)
library(statnet)
library(intergraph)
g <- read.graph('../../examples/bipartite/simulated/bpnet_A12000_B4000_attrs_sim830000000.net', format='pajek')
catattr <- read.table('../../examples/bipartite/simulation/catattr_all.txt', header=T)
gn <- asNetwork(g)
gn <- as.network(as.edgelist(gn), bipartite=12000, directed=FALSE)
network::set.vertex.attribute(gn, 'catattrA', catattr$catattrA)
for (exponent in seq(0, 1, 0.1)) {
  print(summary(gn ~ b1nodematch("catattrA", alpha=exponent) + b1nodematch("catattrA", beta=exponent)))
}

