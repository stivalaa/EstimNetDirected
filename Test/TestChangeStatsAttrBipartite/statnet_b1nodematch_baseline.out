b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
                1987                 1542                  247                  247 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1570.548              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1601.731              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1635.822              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1673.129              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1713.994              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1758.803              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1807.989              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
             1987.00              1862.04               247.00               247.00 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
            1987.000             1921.503              247.000              247.000 
b1nodematch.catattrA b1nodematch.catattrA b2nodematch.catattrP b2nodematch.catattrP 
                1987                 1987                  247                  247 
